ui/src/AppCharting/LineChartjs.svelte

```diff
@@ -0,0 +1,223 @@
+<script lang="ts">
+  import Chart from 'chart.js/auto';
+  import {createEventDispatcher} from 'svelte';
+  import {BY_SIZE, classNameFromBenchmarkKey} from './chart-data-extractor';
+  import {DEFAULT_CHART_HEIGHT} from './chart-layout-helpers';
+
+  export let axis;
+  export let data;
+  export let errorTicks;
+  export let form;
+  export let legendTitle: string;
+  export let selectedTimestamps;
+
+  const dispatch = createEventDispatcher<{rendered: void}>();
+  const CHART_ID = 'open-dds-chart';
+
+  let formattedErrors;
+  let key = Date.now();
+  let chartRef: any = null;
+
+  $: scenario = form.scenario;
+  $: chartType = form.chartType;
+
+  $: if (data && data.columns.length > 0 && key) {
+    console.debug('Drawing');
+    // Allow the rest of the UI to update
+    // without blocking to update the chart.
+    drawChart(data, axis);
+  }
+
+  const formatTime = (time: string) => {
+    if (!time.includes('_')) return time;
+
+    const [datePart, timePart] = time.split('_');
+    const [hour, minute, second] = timePart.split('-');
+    return (time = `${datePart} ${hour}:${minute}:${second}`);
+  };
+
+  function formatErrors() {
+    formattedErrors = [];
+    if (!Array.isArray(data.columns) || !Array.isArray(data.columns[0])) return;
+
+    const selectedSet = new Set(selectedTimestamps);
+    const bySize = chartType === BY_SIZE;
+
+    const xLabels = data.columns[0].slice(1);
+
+    const active = [...errorTicks.values()].filter(error => {
+      if (!error.length) return false;
+      const {scenario: es, key} = error[0];
+      return es === scenario && selectedSet.has(key);
+    });
+
+    for (const error_list of active) {
+      for (const error of error_list) {
+        const {key, scenario: _scenario, dateTime, size} = error;
+
+        const timestampAsClassName = classNameFromBenchmarkKey(key);
+
+        const trimmedSize = JSON.stringify(size);
+        const formattedDateTime = dateTime.replace('_', ' ');
+        let className = bySize ? timestampAsClassName : trimmedSize;
+        const label = bySize ? trimmedSize : formattedDateTime;
+        className = formatTime(className);
+        const index = xLabels.indexOf(label);
+        if (index && label && className) {
+          formattedErrors.push({index, label, className});
+        }
+      }
+    }
+  }
+
+  function onRendered(theKey: any): void {
```

Comment:
Can `theKey` really be any kind of value?

Reply:
This was leftover from the C3 refactor, but I changed to `number` since `key` and `theKey` are always `Date.now()`s